Source/WTF/wtf/RetainPtr.h

```diff
@@ -187,6 +197,25 @@ template<typename T> class RetainPtr {
     CFTypeRef m_ptr { nullptr };
 };
 
+template<typename T>
+class RetainPtrGetterRetains {
+public:
+    explicit RetainPtrGetterRetains(RetainPtr<T>& retainPtr)
+        : m_wrappedRetainPtr(retainPtr)
+    { }
+
+    operator T*() { return m_wrappedRetainPtr.internalValuePointer(); }
+
+private:
+    RetainPtr<T>& m_wrappedRetainPtr;
+};
+
+template<typename T>
+inline RetainPtrGetterRetains<T> adoptOutParam(RetainPtr<T>& retainPtr)
+{
+    return RetainPtrGetterRetains<T>(retainPtr);
```

Comment:
I don't see the benefit in this returning a `RetainPtrGetterRetains<T>` instead of a `T*`. If this was a friend function instead of a friend class, and returned `retainPtr.internalValuePointer()`, then it would work the same, but just fewer lines of code. In fact, I think we could go further and refactor and merge `internalValuePointer` in here; if this is a friend it has access to `RetainPtr::m_ptr` and could do the same assertions and access directly. Nothing that a member function can do that a friend function can’t.

The division of labor seems a little strange here. It seems to me that `adoptOutParam` is the function that should be asserting that the passed in pointer is null and not `internalValuePointer`; if we merged them then that distinction wouldn’t matter.